pallets/gear/src/manager/journal.rs

```diff
@@ -155,22 +158,25 @@ where
             let _ = TaskPoolOf::<T>::delete(bn, ScheduledTask::PauseProgram(id_exited));
 
             match p {
-                Program::Active(program) => {
-                    Self::remove_gas_reservation_map(
-                        id_exited,
-                        core::mem::take(&mut program.gas_reservation_map),
-                    );
-
-                    Self::clean_inactive_program(
-                        id_exited,
-                        program.memory_infix,
-                        value_destination,
-                    );
-                }
+                Program::Active(program) => Self::clean_inactive_program(id_exited, program),
                 _ => unreachable!("Action executed only for active program"),
             }
 
             *p = Program::Exited(value_destination);
+
+            // `gr_exit()` argument implies value is sent to destination
+            let id_exited = id_exited.cast();
```

Comment:
this logic should be also applied to failing init, according to what we've been discussing